waveform_editor/gui/editor.py

```diff
@@ -15,21 +15,24 @@ class WaveformEditor(Viewer):
         class_=(Waveform, DerivedWaveform),
         doc="Waveform currently being edited. Use `set_waveform` to change.",
     )
-    code_editor = param.ClassSelector(class_=pn.widgets.CodeEditor)
-    error_alert = param.ClassSelector(class_=pn.pane.Alert, default=pn.pane.Alert())
-    has_changed = param.Boolean(doc="Whether there are unsaved changes in the editor.")
```

Comment:
Why not keep this as a param with `allow_refs=True`?

That's nicer for downstream code, who would otherwise need to do `editor.has_changed.rx.value` to get the value.